sup3r/preprocessing/data_handlers/nc_cc.py

```diff
@@ -213,20 +216,34 @@ def get_clearsky_ghi(self):
         )
 
         cs_ghi = cs_ghi.transpose(*Dimension.dims_3d())
-
         cs_ghi = cs_ghi['clearsky_ghi'].data
+
+        # concatenate multiple years, need to consider leap years explicitly
+        # so decadal timeseries dont get shifted
+        multi_year = []
         if cs_ghi.shape[-1] < len(self.rasterizer.time_index):
-            n = int(
-                da.ceil(len(self.rasterizer.time_index) / cs_ghi.shape[-1])
-            )
-            cs_ghi = da.repeat(cs_ghi, n, axis=2)
+            for year in self.rasterizer.time_index.year.unique():
+                n = (self.rasterizer.time_index.year == year).sum()
+                multi_year.append(cs_ghi[..., :n])
 
-        cs_ghi = cs_ghi[..., : len(self.rasterizer.time_index)]
+        cs_ghi = da.concatenate(multi_year, axis=-1)
+        cs_ghi = cs_ghi[..., :len(self.rasterizer.time_index)]
```

Comment:
Maybe this would be cleaner if we just matched the doy in ``cs_ghi.time`` with the doy in ``self.rasterizer.time_index`` for each element?

Reply:
Very good call. Check this out. Will want to test on large timeseries once kestrel is back up but i think this should work!

https://github.com/NREL/sup3r/pull/249/commits/13b11dcc4c09ce5f83b58dca22fb4ff7c531cf3d

Reply:
Yeah that looks nice. I think you can also just store ``cs_ghi[Dimension.TIME]`` right after coarsening https://github.com/NREL/sup3r/blob/2d33c8b32f6f26f2171457170e2e828dc4610cc9/sup3r/preprocessing/data_handlers/nc_cc.py#L207 to get what you have here https://github.com/NREL/sup3r/blob/2d33c8b32f6f26f2171457170e2e828dc4610cc9/sup3r/preprocessing/data_handlers/nc_cc.py#L220

Reply:
hmmm maybe im misunderstanding but i dont see the benefit of storing the index up at line 207 vs. pulling it at 220 and converting to month-day. I kind of like having everything contained in these few lines that reindex the ti. I should def be using `Dimension.TIME` though. will push that.

Reply:
Yeah, disregard that initial message. At first glance I thought there was some extra unneeded conversion happening. Edited to clarify.

Reply:
Gotcha! I think I saw the time index is stored as a numpy array so needed the to datetime. doesnt hurt anyway.